### PR TITLE
Tabular: Added FastAI NN to default models

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

Adds FastAI NN to default models. Also adds `torch` and `fastai` dependencies to Tabular. Plan is to move these into optional dependencies prior to v0.1 release.

Benchmark results:

`AutoGluon_bestquality_1h_2020_12_25` : Mainline
`AutoGluon_bestquality_fastai_1h_2020_12_30_fai` : This PR

```
AutoGluon_bestquality_1h_2020_12_25 VS all
                                        framework  > AutoGluon_bestquality_1h_2020_12_25  < AutoGluon_bestquality_1h_2020_12_25  = AutoGluon_bestquality_1h_2020_12_25  % Less Avg. Errors Than AutoGluon_bestquality_1h_2020_12_25  time_train_s  metric_error  time_infer_s  loss_rescaled      rank  rank=1_count  rank=2_count  rank=3_count  rank>3_count  error_count
0  AutoGluon_bestquality_fastai_1h_2020_12_30_fai                                     53                                     35                                      6                                          -0.545924             3334.171915  75266.267894     99.742766       0.393617  1.404255            58            41             0             0            1
1             AutoGluon_bestquality_1h_2020_12_25                                      0                                      0                                     95                                           0.000000             3362.935319  75201.088629    116.126489       0.563830  1.595745            36            59             0             0            5

```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tabular/setup.py [hunk 1]
```diff
@@ -52,8 +52,10 @@ def create_version_file():
     'psutil>=5.0.0,<=5.7.0',  # TODO: psutil 5.7.1/5.7.2 has non-deterministic error on CI doc build -  ImportError: cannot import name '_psutil_linux' from 'psutil'
     'scikit-learn>=0.22.0,<0.24',
     'networkx>=2.3,<3.0',
-    'gluoncv==0.9.0',
-    f'autogluon.core=={version}'
+    'gluoncv==0.9.0',  # TODO: v0.1 consider using only minimum required code from gluoncv and drop dependency
+    'torch<2.0',  # TODO: v0.1 make optional
+    'fastai<2.0',  # TODO: v0.1 make optional
```
Comment (gradientsky) [inline tabular/setup.py:57]: nit: I usually use `fastai==1.*`

Comment (Innixma) [reply tabular/setup.py]: any difference from `fastai>=1.0,<2.0`?

Comment (gradientsky) [reply tabular/setup.py]: it's the same

**Review by gradientsky — APPROVED** [review]

**Comment by szha** [comment]
Job PR-842-7 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-842/7/index.html

**Pull request merged**